core/vm/memory.go

```diff
@@ -90,11 +90,12 @@ func (m *Memory) GetPtr(offset, size int64) []byte {
 		return nil
 	}
 
-	if len(m.store) > int(offset) {
+	if len(m.store) >= int(offset+size) {
 		return m.store[offset : offset+size]
+	} else {
+		panic("invalid memory: get out of bound")
```

Comment:
This changes the semantics of the program. Previously, a `GetPtr` with a high offset but zero size would return `nil`, whereas here it `panic`s. We can't just change the semantics like that without properly ensuring it's correct. 

I don't see any point in adding an explicit panic here.

Reply:
`core/vm/common.go`:
```
func calcMemSize64WithUint(off *uint256.Int, length64 uint64) (uint64, bool) {
	// if length is zero, memsize is always zero, regardless of offset
	if length64 == 0 {
		return 0, false
	}
```
So I think this PR would introduce a _severe_ bug

Reply:
It will also panic in that case IMO:

```
xuzhiqiangdeMacBook-Pro:go xuzhiqiang$ go run 1.go
panic: runtime error: slice bounds out of range [:5] with capacity 3

goroutine 1 [running]:
main.main()
	/tmp/go/1.go:7 +0x1d
exit status 2
xuzhiqiangdeMacBook-Pro:go xuzhiqiang$ cat 1.go
package main

import "fmt"
func main() {
	s := make([]byte, 3)

	ss := s[5:5]
	fmt.Println(len(ss))
}

```

Reply:
Oh, my bad actually -- in the clause before this, we do check the `size==0` case. SO this wouldn't introduce a `panic`.